### PR TITLE
[MU4] playing_techniques_annotations

### PR DESCRIPTION
**Pull request by vpereverzev**
Added predefined, strongly typed elements for playing techniques. The very first step to the texts overhaul

image (1)

We'll start to support all these types and consider them from the playback POV

### CODE REVIEW

File: src/engraving/libmscore/property.cpp [hunk 1]
```diff
@@ -387,6 +387,8 @@ static constexpr PropertyMetaData propertyList[] = {
 
     { Pid::PREFER_SHARP_FLAT,       true,  "preferSharpFlat",       P_TYPE::INT,            DUMMY_QT_TR_NOOP("propertyName", "prefer sharps or flats") },
 
+    { Pid::PLAY_TECH_TYPE,          true,  "playTechType",          P_TYPE::PLAYTECH_TYPE,            DUMMY_QT_TR_NOOP("propertyName", "prefer sharps or flats") },
```
[the remark this answers is not in the document]
Comment (vpereverzev) [reply src/engraving/libmscore/property.cpp]: yep, thanks

File: src/engraving/types/types.h [hunk 2]
```diff
@@ -489,6 +489,44 @@ enum class TextStyleType {
...
+enum class AnnotationCategory {
+    Undefined = -1,
+    TempoAnnotation,
+    PlayingAnnotation,
+    Other,
+};
+
+enum class PlayingTechniqueType {
+    Undefined = -1,
+    Natural,
```
Comment (Jojo-Schmitz) [inline src/engraving/types/types.h:501]: Is that meant to be Arco?

Comment (vpereverzev) [reply src/engraving/types/types.h:501]: yes

Comment (Jojo-Schmitz) [reply src/engraving/types/types.h:501]: Why not naming it that way?

File: src/engraving/types/typesconv.cpp [hunk 3]
```diff
@@ -807,3 +807,29 @@ DurationType TConv::fromXml(const QString& tag, DurationType def)
 {
     return findTypeByXmlTag<DurationType>(DURATION_TYPES, tag, def);
 }
+
+static const std::vector<Item<PlayingTechniqueType> > PLAY_TECH_TYPES = {
+    { PlayingTechniqueType::Undefined,           "undefined" },
+    { PlayingTechniqueType::Natural,             "natural" },
```
[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply src/engraving/types/typesconv.cpp:813]: Esp. as this IMHO needs to match the channel name in instrument.xml

**Review by RomanPudashkin — APPROVED** [review]

File: src/engraving/types/typesconv.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (cbjeukendrup) [reply src/engraving/types/typesconv.cpp:813]: I believe that these are the values used in XML. 
And wasn't the idea of channels going to be removed/replaced in MU4?

[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply src/engraving/types/typesconv.cpp:813]: Well, 'natural' is definitly not used, yet.

[the remark this answers is not in the document]
Comment (vpereverzev) [reply src/engraving/types/typesconv.cpp:813]: it doesn't need to be translatable as well as it doesn't need to be matched with 'channels' which is redundant legacy functionality

[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply src/engraving/types/typesconv.cpp:813]: OK. However: I still don't like "arco" to hide behind "natural"

**Comment by Asmatzaile** [comment]
Shouldn't it be _détaché_ and _martelé_?
Also, regarding arco, as pointed out by @Jojo-Schmitz, it's not the same as _nat._ or _naturale_. _Naturale_ returns the instrument to its natural or default state and can be applied to all types of instruments, while _arco_ is normally used after _pizz._ to refer to a change in the excitator in bowed instruments (and not necessarily reset other changed parameters), so for me it makes sense that they are two different playing techniques (plus other notation software displays them as such).

**Pull request merged**